pkg/controller/events/task_execution.go

```diff
@@ -0,0 +1,79 @@
+package events
+
+import (
+	"context"
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/flyteorg/flyteidl/clients/go/events"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/event"
+	"github.com/flyteorg/flytepropeller/pkg/controller/config"
+	"github.com/flyteorg/flytestdlib/promutils"
+	"github.com/flyteorg/flytestdlib/storage"
+	"github.com/golang/protobuf/proto"
+)
+
+//go:generate mockery -all -output=mocks -case=underscore
+
+// Recorder for Task events
+type TaskEventRecorder interface {
```

Comment:
do we have outputs for task and  nodeS?

Reply:
yup https://github.com/flyteorg/flyteidl/pull/205